@types/index.d.ts

```diff
@@ -1,65 +1,66 @@
 /// <reference types="typescript" />
 
 declare module "bard-ai" {
-	export interface Session {
-		baseURL: "https://bard.google.com";
-		headers: {
-			"X-Same-Domain": "1";
-			"User-Agent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36";
-			"Content-Type": "application/x-www-form-urlencoded;charset=UTF-8";
-			Origin: "https://bard.google.com";
-			Referer: "https://bard.google.com/";
-			Cookie: string;
-		};
+	declare enum EResponseType {
+		JSON = "json",
+		MD = "markdown",
 	}
 
-	let session: Session;
-	let SNlM0e: string;
+	export type TImage = `${string}.${"jpeg" | "jpg" | "png" | "webp"}` | Buffer;
 
-	export type images = Array<{
-		tag: string;
-		url: string;
-	}>;
-
-	export type IdsT = {
+	export type TIds = {
+		_reqID: string;
 		conversationID: string;
 		responseID: string;
 		choiceID: string;
-		_reqID: string;
-	};
-	export declare function init(sessionID: string): Promise<string>;
+	}
+
+	export type TAskConfig = Partial<{
+		format: `${EResponseType}`;
+		image: TImage;
+		ids: Partial<TIds>,
+	}>;
 
-	export type queryBardValidRes = {
+	export type TBardConfig = Partial<{
+		verbose: boolean;
+		fetch: typeof fetch;
+		context: string;
+	}>
+
+	export interface IAskResponseJSON {
 		content: string;
-		images: Array<{
+		images: {
 			tag: string;
 			url: string;
-		}>;
-		ids: IdsT;
-	};
-	export declare function queryBard(
-		message: string,
-		ids?: IdsT | Record<string, string>
-	): Promise<queryBardValidRes | string>;
-
-	export type formatMarkdown = (text: string, images: images) => string;
+			source: {
+				original: string;
+				website: string;
+				name: string;
+				favicon: string;
+			}
+		}[];
+		ids: TIds;
+	}
 
-	export declare function askAI(
-		message: string,
-		useJSON?: boolean
-	): Promise<queryBardValidRes | undefined | string>;
+	declare class Chat {
+		ids?: TIds;
 
-	export declare class Chat {
-		ids?: IdsT | Record<string, string>;
-		constructor(ids?: IdsT | Record<string, string>);
-		ask(message: string, useJSON?: boolean): Promise<queryBardValidRes | string>;
+		ask(message: string, config?: TAskConfig): Promise<IAskResponseJSON | string>;
 		export(): typeof this.ids;
 	}
 
-	declare const Bard: {
-		askAI: typeof askAI;
-		init: typeof init;
-		Chat: typeof Chat;
+	declare class Bard {
+		static JSON = EResponseType.JSON;
+		static MD = EResponseType.MD;
+
+		SNlM0e?: string;
+
+		cookie: string;
+
+		constructor(cookie: string, config?: TBardConfig)
+
+		ask(message: string, config?: TAskConfig): Promise<IAskResponseJSON | string>;
```

Comment:
TypeScript has method overloading. This could be rewritten as
```suggestion
ask(message: string): Promise<string>;
ask(message: string, config: TAskConfig & {format: EResponseType.JSON}): Promise<IAskResponseJSON>;
```
(Although I'm not too experienced with TypeScript so please double-check my syntax)

Reply:
I have already thought about this, I also managed to make it work but there is one problem, its the intellisense will only recognize one of the two formats. I might be missing something, will look after this later on...